### PR TITLE
[full-ci] Redesign New- and Upload-Buttons

### LINKED ISSUE #6279: Redesign New-Button (v1)
# Description

## User Stories
**v1 - must have:**
* > As a user I expect the "New" button to look as nice as the rest of the new design ;-)

**v2 - nice to have**
* > As a user who wants to create a new file I want a clear New-Button-Drop-Menu so that I can find the right filetype quickly. 

* > As a user who wants to upload a resource I want a straight forward way to upload resources so that I need only a few clicks for uploading.

## Value
**v2 - nice to have**
- more clear New-drop menu
- efficient upload-option
- encourage drag & drop for upload via the visual appearance of the upload button

## Acceptance Criteria (Desktop)
**v1 - must have:**
- Redesign New-Button incl. drop menu according to design (no sub-menus)
- add seperators to group the menu items visually
- clickable area is the whole menu item (not only the text)
Moved here from "redesign polishing (#6280):
- [ ] New & upload buttons: no redundant "new" in the drop menus
- [ ] New & upload buttons: upload button: use generic folder and file icons as they are known from the files list

Note: needs restyling of oc-drop

**v2 - nice to have**
- add "upload" button next to new button
- remove "upload file/folder" option from "new"-Button drop menu
- both buttons can have a drop menu
- if the new button would have "new folder" as the single, only drop menu item, the button has no drop menu and its label is "New folder" with "📂" icon

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Related Issue
- Fixes #6279

## Types of changes
- [X] New feature (non-breaking change which adds functionality)
- [X] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added

### CODE REVIEW

**Comment by pascalwengerter** [comment]
Could use some unit test polishing though, happy about a quick first review

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/22786/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22786/alice-shares-file-to-brian-alice-2022-2-15-01-38-02.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22786/alice-shares-file-to-brian-brian-2022-2-15-01-38-06.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/22797/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22797/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-2-15-03-18-04.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22797/alice-shares-file-to-brian-alice-2022-2-15-03-19-15.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22797/alice-shares-folder-with-file-to-brian-alice-2022-2-15-03-20-26.zip 


</p></details>

File: packages/web-app-files/src/components/ActionMenuItem.vue [hunk 1]
```diff
@@ -14,6 +14,13 @@
         alt=""
         class="oc-icon oc-icon-m"
       />
+      <oc-img
+        v-else-if="action.icon && action.icon.startsWith('https://')"
```
Comment (kulmann) [inline packages/web-app-files/src/components/ActionMenuItem.vue:18]: What about a regex match for `/^https?:\/\//i` so catch both `http://` and `https://`?

File: packages/web-app-files/src/components/AppBar/AppBar.vue [hunk 2]
```diff
@@ -30,96 +30,131 @@
...
+                id="new-file-menu-btn"
+                key="new-file-menu-btn-enabled"
+                v-oc-tooltip="newButtonTooltip"
+                :aria-label="newButtonAriaLabel"
+                variation="inverse"
+                :disabled="uploadOrFileCreationBlocked"
+                class="oc-background-primary-gradient"
+              >
+                <oc-icon name="upload" fill-type="line" />
+                <translate>Upload</translate>
```
Comment (pascalwengerter) [inline packages/web-app-files/src/components/AppBar/AppBar.vue:47]: That's a rebase-gone-wrong here 🙄 not sure what exactly went wrong, but on it. Should be the "new" button

File: packages/web-app-files/src/components/ActionMenuItem.vue (hunk 1 above)
Comment (pascalwengerter) [reply packages/web-app-files/src/components/ActionMenuItem.vue:18]: Could be, not sure if it's worth the struggle when https is kinda omnipresent by now?

Comment (kulmann) [reply packages/web-app-files/src/components/ActionMenuItem.vue:18]: IMO it is worth a helper 🤷 I'll take care.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/22845/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-2-17-12-05-25.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-shares-file-to-brian-alice-2022-2-17-12-06-39.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-shares-folder-with-file-to-brian-alice-2022-2-17-12-07-46.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/22845/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-2-17-12-06-54.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-shares-file-to-brian-alice-2022-2-17-12-08-16.zip 

#### npx playwright show-trace https://cache.owncloud.com/owncloud/web/tracing/22845/alice-shares-folder-with-file-to-brian-alice-2022-2-17-12-09-31.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Upload</strong> https://drone.owncloud.com/owncloud/web/22862/38/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L188.png
  

### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/22862/72/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Move</strong> https://drone.owncloud.com/owncloud/web/22862/21/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIOperationsWithFolderShares-accessToSharesFolder_feature-L62.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/22862/68/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L159.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L169.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L199.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L209.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublic2</strong> https://drone.owncloud.com/owncloud/web/22862/37/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L159.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L169.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L199.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L209.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/22862/66/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L188.png
  

### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Upload</strong> https://drone.owncloud.com/owncloud/web/22877/38/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L188.png
  

### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/22877/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesSearch-search_feature-L129.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Move</strong> https://drone.owncloud.com/owncloud/web/22877/21/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIOperationsWithFolderShares-accessToSharesFolder_feature-L62.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/22877/68/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L159.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L169.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L199.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L209.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/22877/72/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublic2</strong> https://drone.owncloud.com/owncloud/web/22877/37/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L159.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L169.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L199.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles_feature-L209.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/22877/66/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L188.png
  

### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L55.png
  

### webUIUpload-upload_feature-L69.png
  


</p></details>

**Review by kulmann — APPROVED** [review]
Awesome 🎨

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/22900/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesSearch-search_feature-L129.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/22900/72/1
The following scenarios passed on retry:
- webUIUpload/uploadFileGreaterThanQuotaSize.feature:12

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/22900/69/1
The following scenarios passed on retry:
- webUIUserJourney/journey1.feature:11

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/22902/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesSearch-search_feature-L129.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/22905/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesSearch-search_feature-L129.png
  


</p></details>

**Comment by kulmann** [comment]
`webUIFilesSearch/search.feature:129` fails reliably. Needs to be looked at...

**Comment by pascalwengerter** [comment]
> `webUIFilesSearch/search.feature:129` fails reliably. Needs to be looked at...

No clue why but navigating didn't work before the upload happened, changed code in https://github.com/owncloud/web/pull/6358/commits/c17bbce9657d063fe57c5c2d6650c2a5547664a6 to make it pass

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/23047/15/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingAcceptSharesToRoot-acceptShares_feature-L108.png
  


</p></details>

**Comment by pascalwengerter** [comment]
I have some (incomplete) test refactoring locally, but will provide these when working on the subsequent changes in https://github.com/owncloud/web/pull/6351

**Pull request merged**